src/components/Modal.js

```diff
@@ -0,0 +1,159 @@
+    setTitleNote(event.target.value);
+  };
+  const handleContentChange = (event) =>{
+    setContentNote(event.target.value);
+  };
+  const [title, setTitle] = useState('');
+  const [content, setContent] = useState('');
+  const onSubmit = (e) => {
+    e.preventDefault()
+    onAdd({title,content})
+    setTitle('')
+    setContent('')
+  }
+  return (
+    <Grid>
+      <Button 
+        sx={{backgroundColor:'#21b5b4',color:'white',width:200,height:57}} 
+        variant="contained" 
+        onClick={handleClickOpen} 
+        startIcon={<NoteAddIcon />}>
+          Add New Note
+      </Button>
+      <BootstrapDialog
+        onClose={handleClose}
+        aria-labelledby="customized-dialog-title"
+        open={open}
+      >
+        <BootstrapDialogTitle onClose={handleClose}>
+          Add new note
+        </BootstrapDialogTitle>
+        <DialogContent dividers>
+          <form onSubmit={onSubmit}>  
+            <Typography variant='paragraph' component='p' textAlign='center'>
+              <label>Add a new note subject/title</label> <br />
+            </Typography>
+            <TextField 
+              sx={{marginTop:1,backgroundColor:'#6ffbff',width:226}}
+              value={title}
+              onChange = {(e) => setTitle(e.target.value)}
+              required
+              name='title'
+              id="title"
+              label="Required"
+            />
+            <Box sx={{marginTop:2}}>
+              <Typography 
+                variant='paragraph' 
+                component='p' 
+                textAlign='center'>
+              <Box>
+                <label>Add content of note</label> <br />
+              </Box>
+            <TextField 
+              sx={{marginTop:1, backgroundColor:'#6ffbff', width:226}}
+              required
+              value={content}
+              onChange = {(e) => setContent(e.target.value)}
+              name='content'
+              id="content"
+              label="Required"
+              multiline
+              rows={4}
+              inputProps = {{maxLength: 300}}
+            />
+              </Typography><br />
+            </Box>
+            <Stack direction="row" spacing={2}>
+              <Button 
+                sx={{width:103, height:40, backgroundColor:'#0097b3'}} 
+                variant="contained"  
+                disabled={!title||!content} 
+                startIcon={<AddCircleIcon />} 
+                type='submit' 
+                autoFocus>
+                  Add
+              </Button>
+              <Button 
+                sx={{width:103, height:40, backgroundColor:'#0097b3'}} 
+                variant="contained" 
+                onClick={handleClose} 
+                startIcon={<CancelIcon />} 
+                type='reset' 
+                autoFocus>
+                  Cancel
+              </Button>
+            </Stack>
+          </form>
+        </DialogContent>
+      </BootstrapDialog>
+    </Grid>
+  );
+}
```

Comment:
each react component should be in its own file, please create a file for each component, then I will review this logic

Reply:
still valid